### DISCUSSION TITLE
Partial from intersection

### DISCUSSION THREAD

**Discussion by bambang-ap** [colinhacks/zod#2068] [Q&A] [answered]
Hi, how to make omit, partial, etc in zod intersection object?

[here the playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcBEyEAJvgLABQlA9NXACoAWwAznK3CAJ5wDGAhiwCmAGjgB1Rv3hCAjgFdgAN34AbIQDt4MCHBjMWAfkowuYIXACSxOAF5E7YgC44LGFGAaA5ugDcJswsAYXk3XCEoOysbADIHDX4QIRc3D28-APM4AAV+WGA1ELCkyPtc-LUAHiKdEoA+fyoKXggNN0cAZV5GIRB+KOQAOggAIwArIV4YAAokYGcUQdTPL2mASnQ1yha2+F5Q2oiunr6o+ePe-kH+DWJpodGJqdm4BKSXIeXvdc2t5tb2mA8jACqoauEoBdTvZ9sUjt1LoMgRVVD9aHoDHAvBAZFAcFAaHQ4MSSaSyeSKZTyQA9YxNQlwABC8hgYmA8AA7tAANbsTBwVTAbkWfSsTLBA4QgBMUTmCy+Xl8r0SyVc7hWGQopiy5RBhUlJRlZWBoOqBoiUrqlG2APg5wRfSNi0ekxmco+S3V3zWYjeqs+XtWGzQfx2gJN+rhkId-Cd9pOsaREdRf2tWsCEhw3nBJQGg08mAilW1Qgg-ORerB5ujCatFHRxNp4rgQWgUFdOYieYLRZLZbgFdBnZrl0tcHRTA4HA5UjtryEQmIYhGLPYGk4PAEwis88Xa5gEWEU2ArTgahYEAZjcMQA)

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?
https://github.com/colinhacks/zod#extend

```ts
const idSchema = z.object( { id: z.string() } )
const customerSchema = idSchema.extend( { name: z.string() } )
const partialCustomerSchema = customerSchema.partial() // no errors, yay!
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by bambang-ap** [reply]
Wow thanks Jacob, You save my day.
